### ISSUE TITLE
Adding Custom Tools

### ISSUE THREAD

**Issue by darkreader2636** [Kiki79250CoC/SMWMTL#7]
If we can add our tools manually this tool become more useful for other hackers too (like YI hackers)

**Comment by Kiki79250CoC** [comment]
Yeah, this kind of feature is on my mind since one year or two, but everytime i postpone it, as it requires some heavy modifications onto a large part of the application and will cause breaking changes at the settings level, but **it's doable, I just need the time to make this idea a concrete thing**.

For example, I have this unfinished mockup I made 8 months ago, when I was about to release SMWMTL 2.20 RC4, that illustrated what I had in mind when I started to port an more conceptual intention of this feature.

![SMWMTL_TotalModularityConcept](https://user-images.githubusercontent.com/54137141/207844317-70345a55-1056-4e48-8330-335f03ea5630.png)

For the moment, my efforts are ported to the SMWMTLextensions DLL, which will provide some features I need for planned features in future releases, as well as a better dark mode support.

So maybe it will finally implemented in one year or two in a next major release (in this case the change will be so important that it will cause the bump to v3.00).
